src/Analyzer.JsonRuleEngine.UnitTests/EqualsOperatorTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Templates.Analyzer.JsonRuleEngine.UnitTests
+{
+    [TestClass]
+    public class EqualsOperatorTests
+    {
+        [DataTestMethod]
+        [DataRow("value", DisplayName = "String values are equal")]
+        [DataRow(true, DisplayName = "Boolean values are equal")]
+        [DataRow(1, DisplayName = "Integer values are equal")]
+        [DataRow(0.1, DisplayName = "Float values are equal")]
+        [DataRow(new string[] { "value1", "value2" }, DisplayName = "Array values are equal")]
+        [DataRow(@"{""property"": ""value""}", DisplayName = "Json values are equal")]
+        public void EvaluateExpression_PropertyIsEqual_EqualsExpressionIsTrue(object jTokenValue)
+        {
+            var jToken = ToJToken(jTokenValue);
+
+            // {"Equals": jTokenValue} is true
+            var equalsOperator = new EqualsOperator(jToken, isNegative: false);
+            Assert.IsTrue(equalsOperator.EvaluateExpression(jToken));
+        }
+
+        [DataTestMethod]
+        [DataRow("value", "value2", DisplayName = "String values are not equal")]
+        [DataRow(true, false, DisplayName = "Boolean values are not equal")]
+        [DataRow(1, 2, DisplayName = "Integer values are not equal")]
+        [DataRow(0.1, 0.2, DisplayName = "Float values are not equal")]
+        [DynamicData(nameof(TestArrays), DynamicDataSourceType.Method, DynamicDataDisplayName = "GetArrayValuesDynamicDataDisplayName")]
+        [DataRow(@"{""property"": ""value11""}", @"{""property2"": ""value12""}", DisplayName = "Json values are not equal")]
+        public void EvaluateExpression_PropertyIsNotEqual_EqualsExpressionIsFalse(object expectedValue, object actualValue)
```

Comment:
Also add tests for comparing values of different types

Reply:
What is the expectation here? Would it throw an error or simply return that it is not equal? Right now it does the later, and I can add another DataRow.